dolphin/db/sqlalchemy/api.py

```diff
@@ -239,7 +239,12 @@ def storage_create(context, values):
 
 def storage_update(context, storage_id, values):
     """Update a storage device with the values dictionary."""
-    return NotImplemented
+    session = get_session()
+    with session.begin():
+        query = storage_get(context, storage_id)
+        query.update(values)
+        query.save(session=session)
+    return
```

Comment:
1. we can reuse the session here when query the storage
2. need to return the result after update values
logic under session.begin may look like:
```
query = _storage_get_query(context, session)
result = query.filter_by(id=storage_id).update(values)
if not result:
    raise exception.StorageNotFound(id=storage_id)
```

Reply:
done